src/en/reference-hook-tools.md

```diff
@@ -0,0 +1,564 @@
+start to return false.
+
+### leader-set
+
+Every service deployed by Juju has access to a pseudo-relation over which
+leader settings can be communicated. 
+
+`leader-set` acts much like [`relation-set`](#relation-set), in that it lets 
+you write string key/value pairs (in which an empty value removes the key), but 
+with the following differences:
+
+      * there's only one leader-settings bucket per service (not one per unit)
+      * only the leader can write to the bucket
+      * only minions are informed of changes to the bucket
+      * changes are propagated instantly, bypassing the sandbox
+
+The instant propagation may be surprising, but it exists to satisfy the use case
+where shared data can be chosen by the leader at the very beginning of (say)
+the install hook. By propagating it instantly, any running minions can make
+use of the data and progress immediately, without having to wait for the
+leader to finish its hook.
+
+It also means that you can guarantee that a successful `leader-set` call has
+been reflected in the database, and that all minions will converge towards
+seeing that value, even if an unexpected error takes down the current hook.
+
+For both these reasons it is strongly recommended that leader settings are 
+always written as a self-consistent group (`leader-set foo=bar baz=qux ping=pong`,
+rather than `leader-set foo=bar; leader-set baz=qux` etc, to avoid situations
+where minions may end up seeing a sandbox in which only `foo` is set to the 
+"correct" value).
+
+### leader-get
+
+`leader-get` acts much like relation-get, in that it lets you read string
+values by key (and expose them in helpful formats), but it reads only from the 
+single leader-settings bucket
+
+As with `realtion-get`, it presents a sandboxed view of leader-settings data. 
+This is necessary, as it is for relation data, because a hook context needs
+to present *consistent* data; but it means that there's a small extra burden
+on users of `leader-set`.
+
+
+### open-port
+
+`open-port` marks a port or range of ports on the local system as appropriate to
+open, if and when the service is exposed to the outside world. It accepts a
+single port or range of ports with an optional protocol, which may be `udp` or
+`tcp`, where `tcp` is the default.
+
+Examples:
+
+Open 80/tcp if and when the service is exposed:
+
+```no-highlight
+open-port 80
+```
+
+Open 1234/udp if and when the service is exposed:
+
+```no-highlight
+open-port 1234/udp
+```
+
+Open the range 8000 to 8080:
+
+```no-highlight
+open 8000-8080/tcp
+```
+
+`open-port` will not have any effect if the service is not exposed, and may have
+a somewhat delayed effect even if it is. This operation is transactional, so 
+changes will certainly not be made unless the hook exits successfully.
+
+Juju also tracks ports opened across the machine and will not allow conflicts - 
+if another charm has already opened the port 
```

Comment:
I believe that is a dash indicating a supplementary clause, however, I generally prefer to employ the ';' where more appropriate, as here.